### PR TITLE
fix: Fixing the pylint error on the loggers.

### PR DESCRIPTION
**Pull request by noble-varghese**
# Description
1. **Pylint Error in Debug Loggers**:
The debug loggers in ther codebase have been flagged with the `Pylint(W1203:logging-fstring-interpolation)` and `Pylint(C0103:invalid-name)`. This error suggests that the names used for variables or functions in the debug loggers do not adhere to the recommended naming conventions.

2. **Inappropriate Usage of f-strings in Debug Loggers**:
While inspecting the debug loggers, it was observed that f-strings were used instead of lazy loading using the %s format specifier. This practice goes against the coding standards and is not considered a best practice for string formatting in loggers. To ensure consistency and maintain code quality, it is necessary to replace the f-strings with the proper usage of the %s format specifier.

3. **Snake Case Conformations in Exception Handling**:
Additionally, some inconsistencies were found in the snake case conformations used for exception handling. We need to review and update the exception handling sections to ensure uniformity and adherence to snake case conventions.

Fixes # (issue)  #6533

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
- [x] I ran the tests on the local and ensured it passes all the test cases in pytest.
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: llama_index/indices/knowledge_graph/retriever.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def _retrieve(
                 rel_map = self._graph_store.get_rel_map(
                     list(subjs), self.graph_store_query_depth
                 )
-                logger.debug(f"rel_map: {rel_map}")
```
Comment (noble-varghese) [inline llama_index/indices/knowledge_graph/retriever.py:162]: %s not added in the debugger.

File: llama_index/indices/knowledge_graph/retriever.py [hunk 2]
```diff
@@ -191,9 +191,9 @@ def _retrieve(
                 embedding_ids=all_rel_texts,
             )
             logger.debug(
-                f"Found the following rel_texts+query similarites: {str(similarities)}"
+                "Found the following rel_texts+query similarites: %s", str(similarities)
             )
-            logger.debug(f"Found the following top_k rel_texts: {str(rel_texts)}")
```
Comment (noble-varghese) [inline llama_index/indices/knowledge_graph/retriever.py:196]: %s not added in the debugger.

**Comment by Disiok** [comment]
Hey @noble-varghese , thanks for the cleanup! Just curious, which pylint configuration results in these errors that you are trying to fix?

**Comment by noble-varghese** [comment]
@Disiok  These are more of warnings than errors. This is from the default configuration provided by pylint and the error is Pylint(W1203:logging-fstring-interpolation) for the debug loggers. 
Hope that answers the question.

**Pull request closed without merging**